tests/ui/components/filters/sortoptionscomponent.js

```diff
@@ -182,4 +183,60 @@ describe('sort options component', () => {
     const wrapper = mount(component);
     expect(wrapper.text()).toEqual('');
   });
+
+  it('uses the persisted sortBys on load', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    const setSortBys = jest.fn();
+    COMPONENT_MANAGER.core.setSortBys = setSortBys;
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [ threeOptions[1] ]);
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(setSortBys).toHaveBeenCalledTimes(0);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(2).getDOMNode().checked).toBeTruthy();
+  });
+
+  it('ignores persisted sortBys that do not match any options from config', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [{
+      type: 'fake type!'
+    }]);
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(0).getDOMNode().checked).toBeTruthy();
+  });
+
+  it('sets persisted sort option on back navigation', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(0).getDOMNode().checked).toBeTruthy();
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [threeOptions[2]]);
+    COMPONENT_MANAGER.core.storage.set(StorageKeys.HISTORY_POP_STATE, {});
```

Comment:
Can you expand on how this properly simulates a back nav? Is updating `HISTORY_POP_STATE` sufficient to fully simulate it? Would an acceptance test be a better fit for this? For a back nav test, I would expect the test to check the option on one page, change the page, and then hit the back button and check again

Reply:
Sure, basically the SortOptions component updates itself on backnav by listening for a HISTORY_POP_STATE event. It doesn't do anything else on back nav. SortOptions is a lot simpler than Facets/FilterBox/FilterOptions, and doesn't interact with other components in weird ways, so I thought a regular jest test would be enough here.

Reply:
okay, that makes sense. I guess I'm still trying to figure out why this test doesn't have three expects: One for the initial page, a second after navigating away to another page, and a third after hitting the back button. And shouldn't the first and third expect have the same option selected?

Reply:
Actually, I see now. All we really have to do in this test is to ensure that it responds to the history pop state change. What do you think about renaming the test to something like "uses the persisted sorts when the history pop state updates"?

Reply:
This test is just for back navigating from a page. I guess to be a little more accurate, this isn't really for "back navigation" so much as "when HISTORY_POP_STATE is updated, reset the component state using persisted storage". I can add another case that tests going back and forth